sdk/identity/azure-identity/TROUBLESHOOTING.md

```diff
@@ -1,218 +1,248 @@
-The `CredentialUnavailableError` indicates that the credential cannot execute in the current environment setup due to lack of required configuration.
-The `ClientAuthenticationError` indicates that the credential was able to run/execute but ran into an authentication issue from the server's end. This can happen due to invalid configuration/details passed in to the credential at construction time.
-This troubleshooting guide covers mitigation steps to resolve these exceptions thrown by various `TokenCredential` implementations in the Azure Identity Python client library.
+This troubleshooting guide covers failure investigation techniques, common errors for the credential types in the Azure Identity Python client library, and mitigation steps to resolve these errors.
 
 ## Table of contents
 
-- [Troubleshooting Default Azure Credential Authentication Issues](#troubleshooting-default-azure-credential-authentication-issues)
-- [Troubleshooting Environment Credential Authentication Issues](#troubleshooting-environment-credential-authentication-issues)
-- [Troubleshooting Service Principal Authentication Issues](#troubleshooting-service-principal-authentication-issues)
-- [Troubleshooting Username Password Authentication Issues](#troubleshooting-username-password-authentication-issues)
-- [Troubleshooting Managed Identity Authentication Issues](#troubleshooting-managed-identity-authentication-issues)
-- [Troubleshooting Visual Studio Code Authentication Issues](#troubleshooting-visual-studio-code-authentication-issues)
-- [Troubleshooting Azure CLI Authentication Issues](#troubleshooting-azure-cli-authentication-issues)
-- [Troubleshooting Azure Powershell Authentication Issues](#troubleshooting-azure-powershell-authentication-issues)
-
-## Troubleshooting Default Azure Credential Authentication Issues
-
-### Credential Unavailable Error
-
-The `DefaultAzureCredential` attempts to retrieve an access token by sequentially invoking a chain of credentials. The `ClientAuthenticationError` in this scenario signifies that all the credentials in the chain failed to retrieve the token in the current environment setup/configuration. You need to follow the configuration instructions for the respective credential you're looking to use via `DefaultAzureCredential` chain, so that the credential can work in your environment.
-
-Please follow the configuration instructions in the `Credential Unavailable Error` section of the troubleshooting guidelines below for the respective credential/authentication type you want to use via `DefaultAzureCredential`:
+- [Handling Azure Identity Errors](#handling-azure-identity-errors)
+  - [ClientAuthenticationError](#clientauthenticationerror)
+  - [CredentialUnavailableError](#credentialunavailableerror)
+  - [Permission Issues](#permission-issues)
+- [Finding Relevant Information in Error Messages](#finding-relevant-information-in-error-messages)
```

Comment:
nit: it would probably be more consistent to use the same tense in these headers as in the rest:
```suggestion
- [Handle Azure Identity Errors](#handle-azure-identity-errors)
  - [ClientAuthenticationError](#clientauthenticationerror)
  - [CredentialUnavailableError](#credentialunavailableerror)
  - [Permission Issues](#permission-issues)
- [Find Relevant Information in Error Messages](#find-relevant-information-in-error-messages)
```